fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/indexes/MultidimensionalIndexMaintainer.java

```diff
@@ -111,7 +118,8 @@ public RecordCursor<IndexEntry> scan(@Nonnull final IndexScanBounds scanBounds,
         final ExecuteProperties executeProperties = scanProperties.getExecuteProperties();
         final ScanProperties innerScanProperties = scanProperties.with(ExecuteProperties::clearSkipAndLimit);
         final CursorLimitManager cursorLimitManager = new CursorLimitManager(state.context, innerScanProperties);
-        final Subspace indexSubspace =  getIndexSubspace();
+        final Subspace indexSubspace = getIndexSubspace();
+        final Subspace secondarySubspace = getSecondarySubspace();
```

Comment:
We may want to add an (additional) prefix to the secondary subspace (before the grouping key). That way, if we find out that there's some other data structure that it would be useful to record about multi-dimensional indexes, we have another place to put it, whereas the current approach uses the full secondary subspace for the node slot index (if I'm understanding its usage correctly)

Reply:
So leave the management for additional tertiary subspaces up to the maintainer itself? I guess that's fine but it is not done by ranked sets currently.

Reply:
Yeah, that is more-or-less what I was suggesting--that the index maintainer would have a set list of tertiary subspaces that it could manage. As you say, that is not what the rank index maintainer does, though I suspect that if we were to do it again without worrying about data compatibility we'd more seriously consider that approach. It may not be super easy to tell, but that is more-or-less how the `TimeWindowLeaderboard` index works, with the secondary subspace carved up a bit to allow multiple types of data to co-exist within it: https://github.com/FoundationDB/fdb-record-layer/blob/bdb3b45d2bfe3bc70e0555533f14f78f7095cd22/fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/leaderboard/TimeWindowLeaderboardIndexMaintainer.java#L137 Lucene plays a similar trick, though it does that even on the primary subspace

Reply:
done.